packages/sdk/src/Base.ts

```diff
@@ -373,36 +388,27 @@ class Base {
     chain: TChain,
     signer: TProvider = this.signer as Signer
   ) {
-    // console.log('getSignerOrProvider')
     chain = this.toChainModel(chain)
     if (!signer) {
       return chain.provider
     }
+
     if (Signer.isSigner(signer)) {
       if (signer.provider) {
         const connectedChainId = await signer.getChainId()
-        // console.log('connectedChainId: ', connectedChainId)
-        // console.log('chain.chainId: ', chain.chainId)
         if (connectedChainId !== chain.chainId) {
-          if (!signer.provider) {
-            // console.log('connect provider')
-            return (signer as Signer).connect(chain.provider)
```

Comment:
curious, why was this line removed?

Reply:
this block of code shouldn't ever run because it is within the condition set by line 382

Reply:
good catch
i think we should still connect `chain.provider` if `signer.provider` doesn't exist since it's possible to use in a signer without a provider and let the sdk handle connecting provider

Reply:
yes, thank you, very thoughtful. on line [409](https://github.com/hop-protocol/hop/blob/796bdf3d106b521579f998a86a7fcd718bdc426c/packages/sdk/src/Base.ts#L409), we connect `chain.provider` as a fallback

Reply:
This should return the signer with connected provider instead of just provider, otherwise transaction won't be able to be made with just the provider

```typescript
return signer.connect(chain.provider)
```

Reply:
ahhhh, i see. thanks for clarifying